### PR TITLE
feat(organizations): add commands for listing, creating, updating, and deleting organizations

### PR DESCRIPTION
**Pull request by rexxars**
### Description

Adds a new `organizations` topic to the CLI with five commands for managing organizations:

- `sanity organizations list` - list all organizations you're a member of
- `sanity organizations get <organizationId>` - show details of a specific organization
- `sanity organizations create` - create a new organization (interactive or `--name`/`--default-role` flags)
- `sanity organizations update <organizationId>` - update name, slug, or default role
- `sanity organizations delete <organizationId>` - delete an organization (with confirmation prompt)

Hidden topic aliases so all of these also work: `organization`, `organisations`, `organisation`, `org`, `orgs`. Uses the two-layer alias approach from #714 (`hiddenAliases` on each command class + `topicAliases.ts` entry).

### What to review

- **`src/services/organizations.ts`** - extended with `getOrganization`, `updateOrganization`, `deleteOrganization`; `createOrganization` now accepts optional `defaultRoleName`; fixed `ProjectOrganization.slug` to be `string | null`
- **`src/commands/organizations/`** - five command files, all thin: parse flags, call service, format output
- **`src/util/apiError.ts`** - shared `hasStatusCode` type guard to avoid `as` assertions in catch blocks
- **`src/topicAliases.ts`** and **`oclif.config.js`** - alias wiring for the `organizations` topic
- **`src/commands/organizations/__tests__/`** - 23 tests across all five commands

### Testing

- 23 unit tests covering all five commands (success, error, edge cases)
- Tests use module-level client mocking (`getGlobalCliClient`), no service file mocks
- Success tests use `if (error) throw error`; error tests assert `toBeInstanceOf(Error)` + message
- All tests pass locally

### Notes

- `--slug` flag only on `update`, not `create` - the API requires the `authSAML` feature for slugs, which a brand-new org won't have
- `delete` blocks server-side if the org has associated projects - the command surfaces that API error clearly
- Members management commands (`organizations members list/invite`) are deferred pending internal discussion

FIXES AIGRO-4968

---

> [!NOTE]
> **Medium Risk**
> Introduces destructive org deletion (with optional `--force`) and account-level PATCH operations; mitigated by name confirmation by default and existing authenticated API client usage.
> 
> **Overview**
> Adds a new **`organizations`** CLI topic (registered in oclif config) with **`list`**, **`get`**, **`create`**, **`update`**, and **`delete`** commands that call the Organizations API via extended helpers in `organizations.ts` (`getOrganization`, `updateOrganization`, `deleteOrganization`, plus optional `defaultRoleName` on create). Users can manage orgs from the terminal with flags or prompts; **`update`** validates name/slug (`validateOrganizationSlug` is new) and supports **`--default-role`**; **`delete`** requires typing the org name unless **`--force`** is used.
> 
> Topic and command aliases are wired through **`topicAliases`** and per-command **`hiddenAliases`** (`org`, `organisation`, etc.). **`promptForOrganizationName`** now works without a user argument and trims input; **`ProjectOrganization.slug`** is typed as nullable. Command behavior is covered by new unit tests and a shared **`httpError`** test helper.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 137e8d752b5549b9d1692c1941b04b64936cddaf. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by runeb — CHANGES_REQUESTED** [review]
Lets make sure `create` returns an actionable error if ran without the required params in non-interactive mode. Currently I think it will just halt with a prompt.

Instead of introducing `hasStatusCode` can we leverage the existing `isHttpError` from `@sanity/client` already in use in the project?

It would also be great if list/get supports `--json` like other commands do, but that is a nicety. I dont think all commands support this generally yet.

**Comment by rexxars** [comment]
Addressed your PR feedback. Yes, `--json` would be nice but we need a unified strategy before we start integrating into each command, IMHO. Eg is there a root shape, or does every command decide their own format? So will leave that for a future iteration. 

Also heading on vacation for a bit, so handing this off to someone else until I'm back 👋

**Review by runeb — DISMISSED** [review]

File: packages/@sanity/cli/src/util/organizationAliases.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const ORGANIZATION_PREFIXES = ['organization', 'organisations', 'organisation', 'org', 'orgs']
+
+export function organizationAliases(action: string): string[] {
+  return ORGANIZATION_PREFIXES.map((prefix) => `${prefix}:${action}`)
+}
```
[the remark this answers is not in the document]
Comment (rexxars) [reply packages/@sanity/cli/src/util/organizationAliases.ts:5]: Fixed. `organizationAliases` now derives its prefixes from `topicAliases.organizations` instead of a duplicated literal, so the two can no longer drift.

File: packages/@sanity/cli/src/commands/organizations/delete.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            trimmed === orgName.trim().toLowerCase() ||
+            'Incorrect organization name. Ctrl + C to cancel delete.'
+          )
+        },
+      })
+    } catch (error) {
+      const err = error instanceof Error ? error : new Error(`${error}`)
+      deleteOrgDebug(`User cancelled`, err)
+      this.error(`User cancelled`, {exit: 1})
+    }
```
[the remark this answers is not in the document]
Comment (rexxars) [reply packages/@sanity/cli/src/commands/organizations/delete.ts:107]: Fixed. `confirmDeletion` now guards with `isUnattended()` before prompting (mirroring `create`), erroring with a message pointing at `--force` instead of rewriting the `NonInteractiveError` as `User cancelled`.

File: packages/@sanity/cli/src/commands/organizations/delete.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    try {
+      await input({
+        message:
+          'Are you ABSOLUTELY sure you want to delete this organization?\n  Type the name of the organization to confirm delete:',
+        validate: (value) => {
+          const trimmed = value.trim().toLowerCase()
+          return (
+            trimmed === orgName.toLowerCase() ||
+            'Incorrect organization name. Ctrl + C to cancel delete.'
+          )
```
[the remark this answers is not in the document]
Comment (rexxars) [reply packages/@sanity/cli/src/commands/organizations/delete.ts:100]: Already addressed in a37bb67d7: the confirmation now compares against `orgName.trim().toLowerCase()`, so names stored with surrounding whitespace pass confirmation. Covered by the 'confirmation matches despite surrounding whitespace' test.

File: packages/@sanity/cli/src/util/apiError.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Type guard for API errors that carry an HTTP status code.
+ */
+export function hasStatusCode(err: unknown): err is {message: string; statusCode: number} {
+  return (
+    typeof err === 'object' &&
+    err !== null &&
+    'statusCode' in err &&
+    typeof (err as Record<string, unknown>).statusCode === 'number'
+  )
```
[the remark this answers is not in the document]
Comment (rexxars) [reply packages/@sanity/cli/src/util/apiError.ts]: Obsolete: `apiError.ts` and the `hasStatusCode` guard no longer exist on this branch. Error handling now uses `isHttpError` from `@sanity/client` and `getErrorMessage` from `@sanity/cli-core/errors`.
